src/iOS/Xamarin.iOS/Samples/Augmented Reality/NavigateAR/RoutePlanner.cs

```diff
@@ -0,0 +1,364 @@
+        {
+            if (_startPoint == null)
+            {
+                // Place the start point.
+                _startPoint = e.Location;
+                Graphic startGraphic = new Graphic(_startPoint, new SimpleMarkerSymbol(SimpleMarkerSymbolStyle.Cross, System.Drawing.Color.Green, 25));
+                _stopsOverlay.Graphics.Add(startGraphic);
+
+                // Update help text.
+                _helpLabel.Text = "Tap to set an end point";
+            }
+            else if (_endPoint == null)
+            {
+                // Place the end point.
+                _endPoint = e.Location;
+                Graphic endGraphic = new Graphic(_endPoint, new SimpleMarkerSymbol(SimpleMarkerSymbolStyle.X, System.Drawing.Color.Red, 25));
+                _stopsOverlay.Graphics.Add(endGraphic);
+
+                // Update help text.
+                _helpLabel.Text = "Solving route";
+
+                // Solve the route.
+                SolveRoute();
+            }
+        }
+
+        private void NavigateButton_Clicked(object sender, System.EventArgs e)
+        {
+            NavigationController.PopViewController(true);
+            NavigationController.PushViewController(new RouteViewerAR() { _routeResult = _routeResult }, true);
+        }
+
+        private async void SolveRoute()
+        {
+            try
+            {
+                // Create the route parameters and configure to enable navigation.
+                _routeParameters = await _routeTask.CreateDefaultParametersAsync();
+                _routeParameters.ReturnStops = true;
+                _routeParameters.ReturnDirections = true;
+                _routeParameters.ReturnRoutes = true;
+
+                // Prefer walking directions if available.
+                TravelMode walkingMode = _routeTask.RouteTaskInfo.TravelModes.FirstOrDefault(mode => mode.Name.Contains("Walk")) ?? _routeTask.RouteTaskInfo.TravelModes.First();
+                _routeParameters.TravelMode = walkingMode;
+
+                // Set the stops.
+                Stop stop1 = new Stop(_startPoint);
+                Stop stop2 = new Stop(_endPoint);
+                _routeParameters.SetStops(new[] { stop1, stop2 });
+
+                // Solve the rotue.
+                _routeResult = await _routeTask.SolveRouteAsync(_routeParameters);
+                _route = _routeResult.Routes.First();
+
+                // Show the route on the map.
+                Graphic routeGraphic = new Graphic(_route.RouteGeometry);
+                _routeOverlay.Graphics.Add(routeGraphic);
+
+                // Update the UI and allow the user to start navigating.
+                _navigateButton.Enabled = true;
+                _helpLabel.Text = "You're ready to start navigating!";
+            }
+            catch (Exception ex)
+            {
+                _helpLabel.Text = "Routing failed, restart sample to retry.";
+                ShowMessage("Failed to calculate route", "Error");
```

Comment:
new UIAlertView("Error", "Failed to calculate route", (IUIAlertViewDelegate) null, "OK", null).Show();